example/main.go

```diff
@@ -2,152 +2,262 @@ package main
+	"crypto/sha256"
+	"encoding/hex"
 	"encoding/json"
 	"fmt"
-	"github.com/Not-Diamond/go-notdiamond/pkg/clients/azure"
-	"github.com/Not-Diamond/go-notdiamond/pkg/clients/openai"
-	"github.com/Not-Diamond/go-notdiamond/pkg/clients/vertex"
 	"io"
 	"log"
 	"net/http"
+	"os"
 	"time"
 
-	"github.com/Not-Diamond/go-notdiamond/pkg/http/response"
-	"github.com/Not-Diamond/go-notdiamond/pkg/transport"
+	"github.com/joho/godotenv"
+)
 
-	"example/config"
+// AWS Signature V4 constants
+const (
+	awsAlgorithm   = "AWS4-HMAC-SHA256"
+	awsRequestType = "aws4_request"
+	awsService     = "bedrock-runtime"
+	awsContentType = "application/json"
+	awsAccept      = "application/json"
+	awsBedrockHost = "bedrock-runtime.%s.amazonaws.com"
+	awsBedrockPath = "/model/%s/invoke"
 )
 
-func main() {
-	// Load configuration
-	cfg := config.LoadConfig()
+// BedrockConfig holds AWS Bedrock configuration
+type BedrockConfig struct {
+	AccessKeyID     string
+	SecretAccessKey string
+	Region          string
+}
+
+// ClaudeRequest represents the request format for Claude models
+type ClaudeRequest struct {
+	AnthropicVersion string                   `json:"anthropic_version"`
+	MaxTokens        int                      `json:"max_tokens"`
+	Messages         []map[string]interface{} `json:"messages"`
+}
+
+// ClaudeResponse represents the response format from Claude models
+type ClaudeResponse struct {
+	Content    []map[string]interface{} `json:"content"`
+	StopReason string                   `json:"stop_reason"`
+	Usage      map[string]int           `json:"usage"`
+}
 
-	// Create API requests
-	openaiRequest, err := openai.NewRequest("https://api.openai.com/v1/chat/completions", cfg.OpenAIAPIKey)
+func main() {
+	// Load .env file
+	err := godotenv.Load(".env")
 	if err != nil {
-		log.Fatalf("Failed to create openai request: %v", err)
+		log.Printf("Warning: Error loading .env file: %v", err)
 	}
-	azureRequest, err := azure.NewRequest(cfg.AzureEndpoint, cfg.AzureAPIKey)
-	if err != nil {
-		log.Fatalf("Failed to create azure request: %v", err)
+
+	// Load configuration from environment variables
+	bedrockConfig := BedrockConfig{
+		AccessKeyID:     os.Getenv("AWS_ACCESS_KEY_ID"),
+		SecretAccessKey: os.Getenv("AWS_SECRET_ACCESS_KEY"),
+		Region:          os.Getenv("AWS_REGION"),
 	}
-	vertexRequest, err := vertex.NewRequest(cfg.VertexProjectID, cfg.VertexLocation)
-	if err != nil {
-		log.Fatalf("Failed to create vertex request: %v", err)
+
+	// Check if Bedrock credentials are available
+	if bedrockConfig.AccessKeyID == "" || bedrockConfig.SecretAccessKey == "" || bedrockConfig.Region == "" {
+		log.Fatalf("AWS Bedrock credentials not found in environment variables")
 	}
 
-	// Get model configuration
-	modelConfig := config.GetModelConfig()
-	modelConfig.Clients = []http.Request{
-		*openaiRequest,
-		*azureRequest,
-		*vertexRequest,
+	fmt.Println("🔑 AWS Bedrock credentials loaded successfully")
+	fmt.Printf("Using AWS region: %s\n", bedrockConfig.Region)
```

Comment:
Nitpick: you can pass multiple params to `fmt.Println` and it'll concatenate them and add a newline, saves a few keys:
```suggestion
	fmt.Println("Using AWS region: ", bedrockConfig.Region)
```